docs/user/reference/hook.md

```diff
@@ -137,6 +142,236 @@ The result is that the events are consumed in the order: `2-1-3`. Beware.
+* $CLOUD_API_VERSION holds the API version of the cloud endpoint.
+* $JUJU_VERSION holds the version of the model hosting the local unit.
+* $JUJU_CHARM_HTTP_PROXY holds the value of the `juju-http-proxy` model config attribute.
+* $JUJU_CHARM_HTTPS_PROXY holds the value of the `juju-https-proxy` model config attribute.
+* $JUJU_CHARM_FTP_PROXY holds the value of the `juju-ftp-proxy` model config attribute.
+* $JUJU_CHARM_NO_PROXY holds the value of the `juju-no-proxy` model config attribute.
+
+Additionally, hooks will have extra environment variables set which correspond to the type of hook.
+Not all environment variables listed are set for every hook of a given type. Where relevant, each
+hook listed below mentions which specific environment variables are set for that particular hook.
+
+### Action hooks
+
+* $JUJU_ACTION_NAME holds the name of the action.
+* $JUJU_ACTION_UUID holds the UUID of the action.
+
+### Relation hooks
+
+These hooks require an associated relation.
+The hook names that these kinds represent will be prefixed by the relation name; for example, `database-relation-joined`.
+
+The following environment variables may be set depending on the exact hook:
+
+* $JUJU_RELATION holds the name of the relation.
+* $JUJU_RELATION_ID holds the ID of the relation.
+* $JUJU_REMOTE_UNIT holds the name of the related unit. 
+* $JUJU_REMOTE_APP holds the name of the related application.
+* $JUJU_DEPARTING_UNIT holds the name of a related unit departing the relation.
+
+### Secret hooks
+
+These hooks require an associated secret.
+The following environment variables may be set depending on the exact hook:
+
+* $JUJU_SECRET_ID holds the ID of the secret.
+* $JUJU_SECRET_LABEL holds the label of the secret.
+* $JUJU_SECRET_REVISION holds the revision of the secret.
+
+### Storage hooks
+
+These hooks require an associated storage.
+The hook names that these kinds represent will be prefixed by the storage name; for example, `database-storage-attached`.
+
+* $JUJU_STORAGE_ID holds the ID of the storage.
+* $JUJU_STORAGE_LOCATION holds the path at which the storage is mounted.
+* $JUJU_STORAGE_KIND holds the kind of storage (`block` or `filesystem`).
+
+### Upgrade series hook
+
+* $JUJU_TARGET_BASE holds the target base for the machine.
+
+### Workload (pebble) hooks
+
+These hooks require an associated workload/container.
+The hook names that these kinds represent will be prefixed by the workload/container name; for example, `mycontainer-pebble-ready`.
+
+The following environment variables may be set depending on the exact hook:
+ 
+* $JUJU_WORKLOAD_NAME holds the name of the container to which the hook pertains.
+* $JUJU_NOTICE_ID holds the Pebble notice ID.
+* $JUJU_NOTICE_TYPE holds the Pebble notice type.
+* $JUJU_NOTICE_KEY holds the Pebble notice key.
+* $JUJU_PEBBLE_CHECK_NAME holds the name of the Pebble check.
+
+## Hook ordering
+
+A charm's lifecycle consists of distinct **phases**:
+* install
```

Comment:
```suggestion
* installation
```

Reply:
Actually, up to you. (Everything else in this list is a noun or can be treated as a noun, except for "install" -- that's why I suggested "installation". But I leave it to your native ear to decide.)

Reply:
You idea makes sense. Fixed.